python/pyspark/sql/dataframe.py

```diff
@@ -1980,10 +1980,16 @@ def dropDuplicates(self, subset=None):
         |Alice|  5|    80|
         +-----+---+------+
         """
+        if isinstance(subset, str):
+            subset = [subset]
```

Comment:
Let's don't include other changes here, and only let the change target the exception improvement alone. It's because strictly we should include other changes like `subset` to `*subset` in the arguments to provide the consistent support with Scala side, more specifically `def dropDuplicates(col1: String, cols: String*)` signature.

For now, let's match the support with `dropDuplicates(colNames: Seq[String])` signature for now.